src/lib/map.c

```diff
@@ -1,2 +1,59 @@
 #include "lib/map.h"
 
+size_t fnv_1a(char *ptr, size_t size) {
+    size_t hash = 0x811C9DC5;
+
+    for (;size--;) {
+        hash ^= ptr[size];
+        hash *= 0x01000193;
+    }
+
+    return hash;
+}
+
+int map_init(struct map *m, size_t size) {
+	m->table = calloc(size, sizeof(struct entry));
```

Comment:
You need to include `stdlib/stdalloc.h` for dynamic memory allocation functions or this will result in a linker error